### ISSUE TITLE
Feature suggestion BlenderBIM: ICDD

### ISSUE THREAD

**Issue by c-mellueh** [IfcOpenShell/IfcOpenShell#4183]
Hi,
i would like to invest some time in adding an importer/exporter for "[Information Container for Linked Document Delivery](https://publications.rwth-aachen.de/record/826311/files/826311.pdf)" (ICDD) [ISO 21597-1:2020]. This would help with linking documents to IFCs.
But before i start doing this i would like to know if this is wished and will be accepted before i start putting a lot of time in it.

**Comment by Moult** [comment]
Go for it! I'm struggling to understand if this is specific to Blender or it is a generically more just an IFC thing which may optionally have a Blender interface.

**Comment by aothms** [comment]
For people interested in gratuitous academic mumbling this is what I had in the back of my mind https://pure.tudelft.nl/ws/portalfiles/portal/104467111/w78_2021_paper_082_1.pdf

**Comment by c-mellueh** [comment]
> Go for it! I'm struggling to understand if this is specific to Blender or it is a generically more just an IFC thing which may optionally have a Blender interface.

it's more of an ifc thing in general. 
If you are interested i can send you the iso 21597 (but only in german).
Is there allready a framework within ifcopenshell that allows you to work with RDF / Ontologies?

**Comment by aothms** [comment]
No, I normally just use rdflib, https://github.com/oxigraph/oxrdflib if you need to query large graphs.

**Comment by Moult** [comment]
There is some work done with Brickschema but this is dealing with Brick's RDF (via rdflib under the hood). BBIM ships with rdflib which allows us to add triples that link Brick to IFC GUIDs. It includes some clever transaction handling too on top.

I'd also note that RDF is extremely slow, especially with reasoning on anything that isn't a tiny model.

There is an IFC TTL converter out there floating around but it isn't very practical.

**Comment by aothms** [comment]
@Moult Give https://github.com/oxigraph/oxrdflib a try, it's a drop in replacement, the default rdflib graph storage is rather naive.

**Comment by c-mellueh** [comment]
The advantage of ICDD is that you don't need to transform your IFC-STEP to an IFC-Onto to connect them.
The ICDD is just an container following specific structural rules.
The ICDD uses RDF-Graphs to define the linkages between files and files/elements in files but doesn't require those files to be a graph itself.
There are 3 main ways to define the connection:
- link to an URI
- link to a GUID/String
- link by queryExpression

for us linking to a GUID/String is the most usefull. 

ICDDs aren't mainly used for reasoning and Web4.0 but more like a standardized way to define a container with links between multiple documents
